### PR TITLE
Medical dart guns can now be emagged

### PR DESCRIPTION
**Pull request by BobJoga**
## About The Pull Request

Title. When emagged they can use any type of syringe as ammo.

## Why It's Good For The Game

Good for medical traitors. 
~~Smart syringe guns were a fucking terrible idea in the first place, but if i do a PR removing them it won't be accepted anyways.~~

## Changelog
:cl:
tweak: Medical dart guns can now be emagged.
/:cl:

### CODE REVIEW

**Comment by necromanceranne** [comment]
dart guns exist for 4 tc, why even add this if not more emag bloat.

**Comment by BobJoga** [comment]
> mini syringe guns exist for 4 tc, why even add this if not more emag bloat.

It's an additional use for your emag. Does it essentially make mini syringe guns a fucking terrible and useless choice? Yes, it does. Just like they were before we added "smart syringe guns".

**Comment by necromanceranne** [comment]
smartdarts are literally just harmless medicinal guns, while a powercreep on old syringe guns for medicinal purposes for sure, they are not quite the same. 

Also, the reason I state this is that an emag is an investment of about 6 Tc but gets you equivalent chems to a poison kit if you have a chem dispenser. For chemists that's not entirely too difficult (short of someone finding your emagged machine), and emags have considerably more robust options beyond this too. They are a powerful tool, but making medical tools into equivalent weapons to something the uplink already provides is not a great idea for viability.

Also mini dart guns are entirely silent and fit in your pocket, what do you mean they're being outpaced by smart dart guns?

**Comment by BobJoga** [comment]
> smartdarts are literally just harmless medicinal guns, while a powercreep on old syringe guns for medicinal purposes for sure, they are not quite the same.
> 
> Also, the reason I state this is that an emag is an investment of about 6 Tc but gets you equivalent chems to a poison kit if you have a chem dispenser. For chemists that's not entirely too difficult (short of someone finding your emagged machine), and emags have considerably more robust options beyond this too. They are a powerful tool, but making medical tools into equivalent weapons to something the uplink already provides is not a great idea for viability.
> 
> Also mini dart guns are entirely silent and fit in your pocket, what do you mean they're being outpaced by smart dart guns?

"They are a powerful tool, but making medical tools into equivalent weapons to something the uplink already provides is not a great idea for viability."
"Also mini dart guns are entirely silent and fit in your pocket, what do you mean they're being outpaced by smart dart guns?"

You kind of killed your own argument there. The mini dart guns still have their place, this is just an additional use for the emag.

File: code/modules/projectiles/guns/misc/syringe_gun.dm [hunk 1]
```diff
@@ -117,12 +117,20 @@
...
+		emag_act(user)
+		return TRUE
 	else
 		to_chat(user, "<span class='notice'>You can't put the [A] into \the [src]!</span>")
 		return FALSE
 
+/obj/item/gun/syringe/dart/emag_act(mob/living/user)
+	to_chat(user, "<span class='warning'>\the [src] will now accept any type of syringe.</span>")
+	playsound(user.loc, "sparks", 50, 1)
+	obj_flags |= EMAGGED
```
Comment (Ghommie) [inline code/modules/projectiles/guns/misc/syringe_gun.dm:132]: It's supposed to return TRUE or FALSE so we know when to detract emag charges or not.
```suggestion
	. = ..()
	if(obj_flags & EMAGGED)
		return
	to_chat(user, "<span class='warning'>\the [src] will now accept any type of syringe.</span>")
	playsound(user.loc, "sparks", 50, 1)
	obj_flags |= EMAGGED
	return TRUE
```

File: code/modules/projectiles/guns/misc/syringe_gun.dm [hunk 2]
```diff
@@ -117,12 +117,20 @@
 	chambered = new /obj/item/ammo_casing/syringegun/dart(src)
 
 /obj/item/gun/syringe/dart/attackby(obj/item/A, mob/user, params, show_msg = TRUE)
-	if(istype(A, /obj/item/reagent_containers/syringe/dart))
+	if(istype(A, /obj/item/reagent_containers/syringe/dart) || (istype(A, /obj/item/reagent_containers/syringe) && obj_flags & EMAGGED))
 		..()
+	else if(istype(A, /obj/item/card/emag))
+		emag_act(user)
+		return TRUE
```
Comment (Ghommie) [inline code/modules/projectiles/guns/misc/syringe_gun.dm]: Redundant, it's already done on `card/emag/afterattack`.
```suggestion
```

**Comment by necromanceranne** [comment]
I certainly haven't because if you consider that we don't have syringe guns roundstart anymore, the emag's charges are 5 per 2TC, and this is only one charge and it makes an innocuous tool and weapon in plain sight, it is actually extremely good value compared to the mini dart gun. That's my point. The mini dart gun is supposed to be a concealable syringe gun but it still marks you as a traitor if someone finds it on your person and that's for 4 TC. The emagged smartdart gun is a weapon in plain sight. And only one of those is getting you executed for having it in your possession and only one of those can be meta'd on sight.

Additionally, the smartdart gun is meant to be relatively 'safe' as far as use goes (ignoring morphine darts). If you add an emag functionality people will valid you over using the smartdart gun. Which means nobody will even use it for fear that they'll get called a traitor or something. Technically the syringe gun used to be for this same purpose. Long range application of medicine but also insidious purposes too. Smartdart was kinda meant to be the 'I can at least know it's not toxins injected into me' alternative.

**Comment by kevinz000** [comment]
Unnecessary change, you can already fuck with the darts to make them take non-medicine chemicals instead of just medical chemicals (last I checked anyways).  I don't feel like we need to introduce emagged smartdart guns the moment syringe guns were killed.

**Comment by DeltaFire15** [comment]
When I checked code, you can only EMP them to be able to OD people, they'll not inject non-medicine either way.

**Comment by Detective-Google** [comment]
The whole point of medical dartguns is that you can do ranged medical application without being instantly lynched because >syringe gun and this would completely kill that. Spend your 4tc on a dart pistol and get on with your life.

**Comment by Putnam3145** [comment]
stealth revert of a PR that was merged like yesterday, nice

**Comment by BobJoga** [comment]
> I certainly haven't because if you consider that we don't have syringe guns roundstart anymore, the emag's charges are 5 per 2TC, and this is only one charge and it makes an innocuous tool a weapon in plain sight, it is actually extremely good value compared to the mini dart gun. That's my point. The mini dart gun is supposed to be a concealable syringe gun but it still marks you as a traitor if someone finds it on your person and that's for 4 TC. The emagged smartdart gun is a weapon in plain sight because having it is a part of your job and the assumption is that it isn't lethal (although going forward paranoia is going to make it basically a whole metagame mess). And only one of those is getting you executed for having it in your possession and only one of those can be meta'd on sight.
> 
> Additionally, the smartdart gun is meant to be relatively 'safe' as far as use goes (ignoring morphine darts). If you add an emag functionality people will valid you over using the smartdart gun. Which means nobody will even use it for fear that they'll get called a traitor or something. Technically the syringe gun used to be for this same purpose. Long range application of medicine but also insidious purposes too. Smartdart was kinda meant to be the 'I can at least know it's not toxins injected into me' alternative.
> 
> What you could do is instead make a disguised syringe gun for chemists, but the issue with that would be that traitor chemists already get the reagent gun, the mini dart gun exists and again invalidated by one that is more openly carriable. Being able to wear weapons in the open is far more powerful than ones you need to keep hidden on your body, just to reiterate my concerns about this.

What if... we make the fucking traitor dart gun not cost 20% of your traitorbucks for something that was basically guaranteed before?

**Comment by BobJoga** [comment]
> stealth revert of a PR that was merged like yesterday, nice

It wasn't a good pr anyways.

**Comment by kevinz000** [comment]
Not interested in a kneejerk revert of a PR that was just merged.
While I have my *thoughts* on the whole removing syringe guns (namely I'm ironically not entirely for it even though I absolutely hate the state of chemical combat), this is the wrong way to go, and I'd rather not have yet another emag-locked function in this specific context. Syringe guns were removed for a reason, this ain't happening right now.

**Pull request closed without merging**